src/spi.rs

```diff
@@ -709,3 +804,132 @@ impl_spi!(SPI3: spi_host_device_t_SPI3_HOST);
 impl_spi_any_pins!(SPI2);
 #[cfg(not(esp32c3))]
 impl_spi_any_pins!(SPI3);
+
+use crate::gpio::{Level, Output};
+use crate::task::CriticalSection;
+use crate::delay::Ets;
+
+pub struct GuardedSpiDevice<'d, DRIVER: Borrow<SpiMasterDriver<'d>>> {
+    device: EspSpiDevice<'d, DRIVER>,
+    mutex: CriticalSection,
+}
+
+impl<'d, DRIVER> GuardedSpiDevice<'d, DRIVER>
+where
+    DRIVER: Borrow<SpiMasterDriver<'d>> + 'd,
+{
+    pub fn new(device: EspSpiDevice<'d, DRIVER>) -> Self {
+        Self {
+            device,
+            mutex: CriticalSection::new(),
+        }
+    }
+}
+pub struct EspSCSSpiDevice<'d, DEVICE, DRIVER: Borrow<SpiMasterDriver<'d>>> {
+    shared_device: DEVICE,
+    cs_pin: PinDriver<'d, AnyOutputPin, Output>,
+    pre_delay: Option<u32>,
+    post_delay: Option<u32>,
+    _p: PhantomData<&'d DRIVER>,
+}
+
+impl<'d, DEVICE, DRIVER> EspSCSSpiDevice<'d, DEVICE, DRIVER>
+where
+    DEVICE: Borrow<GuardedSpiDevice<'d, DRIVER>>,
+    DRIVER: Borrow<SpiMasterDriver<'d>>,
+{
+    pub fn new(
+        shared_device: DEVICE,
```

Comment:
nit: `shared_driver` and perhaps an appropriate rename of the generics from above.

Reply:
done